addons/sale_stock/models/stock.py

```diff
@@ -78,3 +78,51 @@ def _create_backorder(self, backorder_moves=[]):
                     values={'self': backorder, 'origin': backorder.sale_id},
                     subtype_id=self.env.ref('mail.mt_note').id)
         return res
+
+    def _log_less_quantities_than_expected(self, moves):
+        """ Log an activity on sale order that are linked to moves. The
+        note summarize the real proccessed quantity and promote a
+        manual action.
+
+        :param dict moves: a dict with a move as key and tuple with
+        new and old quantity as value. eg: {move_1 : (4, 5)}
+        """
+
+        def _keys_in_sorted(sale_line):
+            """ sort by order_id and the sale_person on the order """
+            return (sale_line.order_id.id, sale_line.order_id.user_id.id)
+
+        def _keys_in_groupby(sale_line):
+            """ group by order_id and the sale_person on the order """
+            return (sale_line.order_id, sale_line.order_id.user_id)
+
+        def _generate_note_missing_quantity(quantity_by_move, order_line_ids, moves_soline_rel):
```

Comment:
instead of the 3 parameters, why not
{'impacted_move':  [('move_orig1': (5, 4)), ('move_orig2': (10, 0))]}